File: src/main/java/uknowlab/designpattern4java/adapter/ex1_plug/PlugAdaptor.java
package uknowlab.designpattern4java.adapter.ex1_plug;

public class PlugAdaptor implements PlugInterface {

    private Plug110V plug110V;

    public PlugAdaptor(Plug110V plug110V) {
        this.plug110V = plug110V;
    }

    @Override
    public void plugIn() {
        plug110V.plugIn110V();
    }
}

File: src/main/java/uknowlab/designpattern4kotlin/adapter/ex1_plug/PlugAdaptor.kt
package uknowlab.designpattern4kotlin.adapter.ex1_plug

import uknowlab.designpattern4java.adapter.ex1_plug.PlugInterface

class PlugAdaptor(private val plug110V: Plug110V) : PlugInterface {
    override fun plugIn() {
        plug110V.plugIn110V()
    }
}
